packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadControls.tsx

```diff
@@ -153,11 +154,24 @@ const renderRowItem: RenderRowItem<LocationActionViewColumns> = (
         );
       case 'cancel':
         if (row.cancel) {
+          // FIXME: Eventually comes from useView hook
+          const contextValue: ControlsContext = {
+            data: {
+              actionCancelAriaLabel: `Cancel upload for ${row.key}`,
+            },
+            actionsConfig: {
+              type: 'SINGLE_ACTION',
+              isCancelable: true,
+            },
+            onActionCancel: row.cancel,
+          };
+
           return (
-            <Cancel
-              onClick={row.cancel}
-              ariaLabel={`Cancel upload for ${row.key}`}
-            />
+            <ControlsContextProvider {...contextValue}>
```

Comment:
It feels weird to wrap every cancel control in each row with a provider.

Reply:
It is all the rows but only all the rows of the Cancel column. The data for ariaLabel and onActionCancel are specific to that column and row. So not sure how I can lift it up.

Reply:
Discussed offline and leaving it as is since row data for this column is only available here.

Reply:
Instead of using this ActionCancelControl here, this could just be converted to using a ButtonElement. I don't think we are going to offer composability at this level.